packages/domain/lib/src/utils/utility_extensions.dart

```diff
@@ -18,6 +18,18 @@ extension NullAbleString on String? {
   String get valueOrEmpty => this ?? '';
 }
 
-extension IsValid on ValidationResult{
+extension IsValid on ValidationResult {
   bool get isFailure => this.login != null || this.password != null;
-}
\ No newline at end of file
+}
+
+extension IsApiRequestAllowed on DateTime? {
+  bool get isApiRequestAllowed {
+    if (this != null) {
+      final currentDate = DateTime.now();
+      return currentDate.year != this!.year &&
+          currentDate.month != this!.month &&
+          currentDate.day != this!.day;
```

Comment:
do u need to write this explicitly in the right part of comparison?

Reply:
I disappointed, but yes, in kotlin null checks works better =(